[FEAT] #45 - 여행 일정 뷰 구현

Taehyeon-Kim opened the pull request:
## 🌴 PR 요약
여행 일정 뷰에 대한 UI 및 기능 구현을 1차 완료했습니다.
피드백 들어온 내용에 대해서는 조치를 취하도록 하겠습니다.

🌱 작업한 브랜치
- feature/#45

🌱 작업한 내용
- 여행 일정 뷰 상단 영역 (캘린더)
- 여행 일정 없을 때 UI
- 여행 일정 하나 추가되었을 때 UI
- 여행 여러 개 추가되었을 때 UI
- 여행 일정 데이터 모델 임시 생성
- 더미데이터 연결

## 📸 스크린샷
|설명|스크린샷|
|:--:|:--:|
|캘린더 영역|<img width="200" alt="스크린샷 2021-07-08 오후 7 34 00" src="https://user-images.githubusercontent.com/61109660/124914140-f6df2700-e02a-11eb-8a81-81c279b32110.gif">|
|일정 없을 때|<img width="250" alt="스크린샷 2021-07-08 오후 7 34 00" src="https://user-images.githubusercontent.com/61109660/124913223-d6fb3380-e029-11eb-9227-c3cd5014a429.png">|
|여행 일정 하나 추가|<img width="250" alt="스크린샷 2021-07-08 오후 7 34 58" src="https://user-images.githubusercontent.com/61109660/124913357-fbefa680-e029-11eb-952c-a94ec28abeab.png">|
|여행 일정 여러 개 추가|<img width="200" alt="스크린샷 2021-07-08 오후 7 36 02" src="https://user-images.githubusercontent.com/61109660/124914173-02cae900-e02b-11eb-802f-6b50bffcd4aa.gif">|


## 📮 관련 이슈
- Resolved: #45

Taehyeon-Kim committed:
[ADD] #45 - 데이터 없을 때 표시할 셀 추가

- 이미지로 내용 대체

Taehyeon-Kim committed:
[ADD] #45 - 여행 일정 셀 UI 

- 시간흐름 ui (top line, point, bottom line)
- 시간
- 일정 제목
- 일정 설명

Taehyeon-Kim committed:
[ADD] #45 - 일정 테이블 뷰 헤더 뷰 UI

- 우선 섹션 헤더에 헤더뷰로 넣는 형태로 구현

Taehyeon-Kim committed:
[ADD] #45 - 여행일정 데이터 모델 생성

- 더미로 우선 생성

Taehyeon-Kim committed:
[FEAT] #45 - 여행일정뷰 테이블뷰 세팅

- 테이블뷰 생성
- 테이블뷰 연결
- 테이블셀 연결
- 각 상황에 따라 셀 다르게 표시

Taehyeon-Kim committed:
[FEAT] #45 - 더미데이터 연결

- 서버로 넘어오기전 더미데이터로 작업
- 테스트
-- 셀 없을 때
-- 셀 하나 있을 때
-- 셀 여러개 있을 때

Taehyeon-Kim committed:
[ADD] #45 - 여행일정뷰 1차 구현 완료

- 추가 피드백 사항 구현하도록 하겠습니다.

Taehyeon-Kim committed:
[FIX] #45 - #51 conflict 해결

##### 제목은 최대 50 글자까지만 입력 ############## -> |

# 본문은 위에 작성

######## 본문은 한 줄에 최대 72 글자까지만 입력 ########################### -> |

# 꼬릿말은 아래에 작성

# --- COMMIT END ---

# <타입> 리스트
# init : 초기화
# add : 파일 추가
# rename : 파일 혹은 폴더명을 수정하거나 옮기는 작업만인 경우
# remove : 파일을 삭제하는 작업만 수행한 경우
# feat : 기능 추가
# delete : 기능 삭제
# update : 기능 수정
# fix : 버그 수정
# refactor: 리팩토링
# style : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
# docs : 문서 (문서 추가(Add), 수정, 삭제)
# chore : 기타 변경사항 (빌드 스크립트 수정, 에셋 추가 등)
# design : 사용자 UI 디자인 변경
# !BREAKING CHANGE : 커다란 API 변경의 경우 !
# hotfix : 급하게 치명적인 버그를 고쳐야하는 경우
# comment : 필요한 주석 추가 및 변경
# test : 테스트 추가, 테스트 리팩토링(프로덕션 코드 변경 X)
# chore : 빌드 태스트 업데이트, 패키지 매니저를 설정하는 경우(프로덕션 코드 변경 X)

# ------------------

# 제목 끝에 마침표(.) 금지
# 제목 끝에는 괄호와 함께 이슈 번호 작성: ex (#이슈 번호)
# 제목과 본문을 한 줄 띄워 분리하기
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명한다.
# 본문에 여러줄의 메시지를 작성할 땐 "-"로 구분

# ------------------

Taehyeon-Kim committed:
[FIX] #45 - 여행일정 첫번째 셀 Top Line 숨김

- indexPath.row 1 -> 0

Hansangjin98 commented:
LGTM...

hyejinL commented on DooRiBon/DooRiBon/Sources/Base/Plan/PlanViewController.swift:
```diff
@@ -170,3 +216,57 @@ extension PlanViewController: UICollectionViewDelegateFlowLayout {
+                return UITableViewCell()
+            }
+            if indexPath.row == 0 {
+                if planDummyData.count == 1 {
+                    cell.bottomLineView.isHidden = true
+                }
+                cell.topLineView.isHidden = true
```
Tip. 아래의 형태도 있습니다. (취향 차이, 권유 아님)
```swift
cell.bottomLineView.isHidden = planDummyData.count == 1
cell.topLineView.isHidden = true
```

hyejinL commented on DooRiBon/DooRiBon/Sources/Base/Plan/PlanViewController.swift:
```diff
@@ -24,14 +24,19 @@ class PlanViewController: UIViewController {
+        }
+    }
+    private var selectedDate: Int?            // 오늘 날짜를 디폴트 값으로 시작
+    private var dataStatus: Bool = false
```
값을 저장할 필요 없이 get property로 해도 될 것 같습니다.
실수로 값을 저장하지 않으면 에러가 발생하니까요.
``` swift 
private var dataStatus: Bool {
    !planDummyData.isEmpty
}
```
혹은 그냥 `planDummyData.isEmpty` 를 사용해도 될 것 같습니다.

mini-min reviewed (approved):
OJA✨

Taehyeon-Kim committed:
[FIX] #45 - dataStatus 변수 get property로 수정

- 실수로 값을 저장하지 않았을 때 에러 발생 가능

Taehyeon-Kim committed:
[REFACTOR] #45 - 더미데이터 3개만 저장

##### 제목은 최대 50 글자까지만 입력 ############## -> |

# 본문은 위에 작성

######## 본문은 한 줄에 최대 72 글자까지만 입력 ########################### -> |

# 꼬릿말은 아래에 작성

# --- COMMIT END ---

# <타입> 리스트
# init : 초기화
# add : 파일 추가
# rename : 파일 혹은 폴더명을 수정하거나 옮기는 작업만인 경우
# remove : 파일을 삭제하는 작업만 수행한 경우
# feat : 기능 추가
# delete : 기능 삭제
# update : 기능 수정
# fix : 버그 수정
# refactor: 리팩토링
# style : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
# docs : 문서 (문서 추가(Add), 수정, 삭제)
# chore : 기타 변경사항 (빌드 스크립트 수정, 에셋 추가 등)
# design : 사용자 UI 디자인 변경
# !BREAKING CHANGE : 커다란 API 변경의 경우 !
# hotfix : 급하게 치명적인 버그를 고쳐야하는 경우
# comment : 필요한 주석 추가 및 변경
# test : 테스트 추가, 테스트 리팩토링(프로덕션 코드 변경 X)
# chore : 빌드 태스트 업데이트, 패키지 매니저를 설정하는 경우(프로덕션 코드 변경 X)

# ------------------

# 제목 끝에 마침표(.) 금지
# 제목 끝에는 괄호와 함께 이슈 번호 작성: ex (#이슈 번호)
# 제목과 본문을 한 줄 띄워 분리하기
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명한다.
# 본문에 여러줄의 메시지를 작성할 땐 "-"로 구분

# ------------------

Taehyeon-Kim committed:
[REFACTOR] #45 - 코드 스타일 수정

- if ~ else -> 한 줄 로

Taehyeon-Kim committed:
[ADD] #45 - pbxproj 누락 파일 추가

##### 제목은 최대 50 글자까지만 입력 ############## -> |

# 본문은 위에 작성

######## 본문은 한 줄에 최대 72 글자까지만 입력 ########################### -> |

# 꼬릿말은 아래에 작성

# --- COMMIT END ---

# <타입> 리스트
# init : 초기화
# add : 파일 추가
# rename : 파일 혹은 폴더명을 수정하거나 옮기는 작업만인 경우
# remove : 파일을 삭제하는 작업만 수행한 경우
# feat : 기능 추가
# delete : 기능 삭제
# update : 기능 수정
# fix : 버그 수정
# refactor: 리팩토링
# style : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
# docs : 문서 (문서 추가(Add), 수정, 삭제)
# chore : 기타 변경사항 (빌드 스크립트 수정, 에셋 추가 등)
# design : 사용자 UI 디자인 변경
# !BREAKING CHANGE : 커다란 API 변경의 경우 !
# hotfix : 급하게 치명적인 버그를 고쳐야하는 경우
# comment : 필요한 주석 추가 및 변경
# test : 테스트 추가, 테스트 리팩토링(프로덕션 코드 변경 X)
# chore : 빌드 태스트 업데이트, 패키지 매니저를 설정하는 경우(프로덕션 코드 변경 X)

# ------------------

# 제목 끝에 마침표(.) 금지
# 제목 끝에는 괄호와 함께 이슈 번호 작성: ex (#이슈 번호)
# 제목과 본문을 한 줄 띄워 분리하기
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명한다.
# 본문에 여러줄의 메시지를 작성할 땐 "-"로 구분

# ------------------

Taehyeon-Kim committed:
Merge branch 'develop' into feature/45-plan-view-bottom

Taehyeon-Kim commented:
피드백 전부 수정했습니다 ✨
